tests/pytests/test_vecsim.py

```diff
@@ -480,7 +481,8 @@ def test_hybrid_query_batches_mode_with_text():
     execute_hybrid_query(env, '(other)=>[KNN 10 @v $vec_param]', query_data, 't').equal(expected_res)
 
     # Expect empty score for the intersection (disjoint sets of results)
-    execute_hybrid_query(env, '(@t:other text)=>[KNN 10 @v $vec_param]', query_data, 't').equal([0])
+    # The hybrid policy changes to ad hoc after the first batch
+    execute_hybrid_query(env, '(@t:other text)=>[KNN 10 @v $vec_param]', query_data, 't', batches_mode=False).equal([0])
```

Comment:
is this an expected change or a by-product of this PR changes?
Also what about the third mode?

Reply:
Its expected that the search mode will be affected here due to the dynamic policy selection. This is a similar use case to the original one - an intersection of two disjoint sets of many results.
The third mode will be validated here...